### ISSUE TITLE
TOC generation plugin generates linting warnings with Markdownlint

### ISSUE THREAD

**Issue by mundodisco8** [shd101wyy/markdown-preview-enhanced#1792]
Hi,

This is a bit of a tough one, because it affects multiple systems that are not necessarily MPE by itself, but I wanted to hear your opinion. My setup is MPE, VS Code and Markdownlint as MD linter.

I create my ToCs with the code snippet execution thingy, like in this doc:

```MD
# Title

## Changelog

| Date | Author | Changes |
| - | - | - |
| 20/01/2021 | Me | First Version of this doc |
| 24/02/2021 | Someone Else | Changed the abstract |

## Table of Contents
<!-- If using Markdown Enhanced Preview in VS Code, the ToC can be autogenerated. Otherwise, do it by hand :S. Press shift+enter in the next line WITH MARKDOWN ENHANCED PREVIEW TAB open in VS Code (making sure SNIPPET EXECUTION IS ENABLED, see this https://shd101wyy.github.io/markdown-preview-enhanced/#/code-chunk) to autogenerate a new ToC -->

<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=2 orderedList=false} -->

<!-- code_chunk_output -->

- [Title](#-title-)
  - [Changelog](#-changelog-)
  - [Table of Contents](#-table-of-contents-)
  - [Introduction](#-introduction-)
  - [Intended Audience](#-intended-audience-)
  - [Content](#-content-)

<!-- /code_chunk_output -->

## Introduction

## Intended Audience

## Content
```

The code follows what I think is Github spec, as described in rule MD051 in Markdownlint ([link](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md)), which replaces spaces with dashes. All good. The thing is Markdownlint is not expecting the dashes between the pound sign from the header and the last dash (which sincerely don't know where it comes from as there are no leftover spaces in my headers at the end of the line.

So, to sum up:

Markdownlint expects the TOC to be:

```MD
- [Title](#title)
  - [Changelog](#changelog)
  - [Table of Contents](#table-of-contents)
  - [Introduction](#introduction)
  - [Intended Audience](#intended-audience)
  - [Content](#content)
```

Instead of the ones on top. I created a ticket on their repo, and they said that the behaviour I see is the expected one. I can live with the MD051 disabled in Markdownlint, but, is it something worth fixing?

thanks!
